log/sequencer.go

```diff
@@ -281,20 +325,34 @@ func (s Sequencer) SequenceBatch(ctx context.Context, logID int64, limit int, gu
 		return 0, s.SignRoot(ctx, logID)
 	}
 
-	// There might be no work to be done. But we possibly still need to create an signed root if the
-	// current one is too old. If there's work to be done then we'll be creating a root anyway.
-	numLeaves := len(leaves)
+	st := &logSequencingTask{
+		label:      label,
+		treeSize:   currentRoot.TreeSize,
+		timeSource: s.timeSource,
+		dequeuer:   tx,
+	}
+	sequencedLeaves, err := st.load(ctx, limit, start.Add(-guardWindow))
+	if err != nil {
+		glog.Warningf("%v: Sequencer failed to load sequenced batch: %v", logID, err)
+		return 0, err
+	}
+	numLeaves := len(sequencedLeaves)
+
+	// There might be no work to be done. But we possibly still need to create a
+	// signed root if the current one is too old. If there's work to be done then
```

Comment:
I think I originally wrote this comment  but it could be made less confusing.

"We need to create a signed root if entries were added or the latest root is too old"

Reply:
Done.